lib/SchemaConnector.js

```diff
@@ -0,0 +1,249 @@
+   *
+   * @param {*} req
+   * @param {*} res
+   */
+  async handleHttpCallback(req, res) {
+    try {
+      const response = await this._handleCallback(req.body);
+      if (response.isError()) {
+        console.log("ERROR: %j", response);
+        res.status(500).send(response)
+      }
+      else {
+        res.send(response)
+      }
+    }
+    catch (err) {
+      console.log("ERROR: %j", err);
+      res.status(500).send(err)
+    }
+  }
+
+  async handleMockCallback(body) {
+    try {
+      return await this._handleCallback(body)
+    }
+    catch (err) {
+      return err;
+    }
+  }
+
+  async _handleCallback(body) {
+    if (this._enableEventLogging) {
+      console.log(`REQUEST ${JSON.stringify(body, null, this._eventLoggingSpace)}`)
+    }
+    if (body.headers) {
+      let response;
+      switch (body.headers.interactionType) {
+        case "discoveryRequest":
+          response = new DiscoveryResponse(body.headers.requestId);
+          await this._discoveryHandler(body.authentication.token, response, body);
+          break;
+
+        case "commandRequest":
+          response = new CommandResponse(body.headers.requestId);
+          await this._commandHandler(body.authentication.token, response, body.devices, body);
+          break;
+
+        case "stateRefreshRequest":
+          response = new StateRefreshResponse(body.headers.requestId);
+          await this._stateRefreshHandler(body.authentication.token, response, body);
+          break;
+
+        case "grantCallbackAccess":
+          response = new STBase(body.headers.requestId);
+          if (this._callbackAccessHandler) {
+            if (body.callbackAuthentication.clientId === this._clientId ) {
+
+              const grantResponse = await (
+                new AccessTokenRequest(
+                  this._clientId,
+                  this._clientSecret,
+                  body.headers.requestId
+                ).getCallbackToken(
+                  body.callbackUrls.oauthToken,
+                  body.callbackAuthentication.code
+                )
+              );
+
+              await this._callbackAccessHandler(body.authentication.token, grantResponse.callbackAuthentication, body.callbackUrls, body)
+            }
+            else {
+              response = new ErrorResponse().setError('Invalid client ID');
+            }
+          }
+          break;
+
+        case "integrationDeleted":
+          response = new STBase(body.headers.requestId);
+          await this._integrationDeletedHandler(body.authentication.token, body);
+          break;
+
+        default:
+          response = new ErrorResponse().setError(`Unsupported interactionType: ${body.headers.interactionType}`);
+          break;
+      }
+
+      if (this._enableEventLogging) {
+        console.log(`RESPONSE ${JSON.stringify(response, null, this._eventLoggingSpace)}`)
+      }
+
+      return response;
+    }
+    else {
```

Comment:
I would flip this around instead of if/then/else:

if (!body.header) {
  return new ErrorResponse().setError(`Invalid ST Schema request. No 'headers' field present.`);
}

let response;
switch (body.headers.interactionType) {
.......

Reply:
:thumbsup: